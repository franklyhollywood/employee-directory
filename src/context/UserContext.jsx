import { useContext, createContext, useState } from 'react';

const userContext = createContext();

const userProvider = ({ children }) => {
  const [user, setUser] = useState('');

  return (
    <userContext.Provider value={{ user, setUser }}>
      {children}
    </userContext.Provider>
  );
};

const useUser = () => {
  const context = useContext(userContext);

  if (context === undefined) {
    throw new Error('Incorrect Placement');
  }
  return context;
};

export { userProvider, useUser };
