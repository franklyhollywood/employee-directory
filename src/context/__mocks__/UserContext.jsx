import { useContext, createContext, useState } from 'react';

const UserContext = createContext();

const UserProvider = ({ mockUser, children }) => {
  const [user, setUser] = useState(mockUser ? { email: mockUser.email } : {});

  return (
    <UserContext.Provider value={{ user, setUser }}>
      {children}
    </UserContext.Provider>
  );
};

const useUser = () => {
  const UserCtx = useContext(UserContext);

  if (UserCtx === undefined) {
    throw new Error('Must be wrapped in a userprovider to use useUser');
  }
  return UserCtx;
};

export { UserProvider, useUser };
