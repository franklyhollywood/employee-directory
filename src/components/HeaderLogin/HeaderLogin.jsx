import { useUser } from '../../context/UserContext.jsx';
import { Link } from 'react-router-dom';
import { signOutUser } from '../../services/users.js';
import { useHistory } from 'react-router-dom';

export default function HeaderLogin({ loggedin }) {
  const { user, setUser } = useUser();
  const history = useHistory();

  const handleLogOut = async () => {
    await signOutUser();
    setUser(null);
    history.push('/login');
  };

  const loggedInDiv = (
    <div>
      {/* ? is elvis notation: checking to see if variable is defined before reading a property of that variable */}
      {user?.email} <button onClick={handleLogOut}>sign out</button>
    </div>
  );

  const notLoggedInDiv = (
    <div>
      <p>Not Signed in</p>
      <Link to="/login">
        <button>Sign in</button>
      </Link>
    </div>
  );

  return <div>{loggedin ? loggedInDiv : notLoggedInDiv}</div>;
}
