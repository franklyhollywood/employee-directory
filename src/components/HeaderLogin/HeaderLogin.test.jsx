import { render } from '@testing-library/react';
import HeaderLogin from './HeaderLogin';
import { MemoryRouter } from 'react-router-dom';
import { UserProvider } from '../../context/UserContext.jsx';

jest.mock('../../services/users.js');
jest.mock('../../context/UserContext.jsx');

it('header test', () => {
  const { container } = render(
    <UserProvider>
      <MemoryRouter>
        <HeaderLogin />
      </MemoryRouter>
    </UserProvider>
  );
  expect(container).toMatchSnapshot();
});
