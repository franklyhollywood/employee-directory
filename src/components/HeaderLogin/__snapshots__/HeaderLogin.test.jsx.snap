// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header test 1`] = `
<div>
  <div>
    <div>
      <p>
        Not Signed in
      </p>
      <a
        href="/login"
      >
        <button>
          Sign in
        </button>
      </a>
    </div>
  </div>
</div>
`;
